src/tag_gc.rs

```diff
@@ -148,10 +156,13 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: MiriInterpCxExt<'mir, 'tcx> {
         }
 
         let mut tags = FxHashSet::default();
-        this.visit_tags(&mut |tag| {
-            tags.insert(tag);
+        let mut alloc_ids = FxHashSet::default();
+        this.visit_tags(&mut |id, tag| {
+            id.map(|id| alloc_ids.insert(id));
+            tag.map(|tag| tags.insert(tag));
```

Comment:
I'm not a fan of calling `map` for side-effects only. Can you make this `if let`, or maybe `.iter().for_each` if you want to optimize for line numbers?